identity-service/src/notifications/trendingTrackProcessing.js

```diff
@@ -40,38 +41,76 @@ const TRENDING_INTERVAL_HOURS = 3
-    let params = new URLSearchParams()
-    params.append('time', TRENDING_TIME.WEEK)
-    params.append('limit', MAX_TOP_TRACK_RANK)
-
-    const { discoveryProvider } = audiusLibsWrapper.getAudiusLibs()
-    const baseUrl = `${discoveryProvider.discoveryProviderEndpoint}/v1/full/tracks/trending`
-    const url = trendingExperiment
-      ? `${baseUrl}/${trendingExperiment}?time=week`
-      : `${baseUrl}?time=week`
-
-    const trendingTracksResponse = await axios({
-      method: 'get',
-      url,
-      params,
-      timeout: 10000
-    })
-    const trendingTracks = trendingTracksResponse.data.data.map((track, idx) => ({
-      trackId: decodeHashId(track.id),
-      rank: idx + 1,
-      userId: decodeHashId(track.user.id)
-    }))
-    const blocknumber = trendingTracksResponse.data.latest_indexed_block
-    return { trendingTracks, blocknumber }
-  } catch (err) {
-    logger.error(`Unable to fetch trending tracks: ${err}`)
+let SELECTED_DISCOVERY_NODES = null
+setInterval(async () => {
+  SELECTED_DISCOVERY_NODES = await getDiscoveryNodes()
+}, 5 * 60 * 60 /* re-run on the 5-minute */)
+
+const getDiscoveryNodes = async () => {
+  const libs = await audiusLibsWrapper.getAudiusLibsAsync()
+  const discoveryNodes = await libs.discoveryProvider.serviceSelector.findAll()
+  logger.debug(`Updating discovery nodes for trendingTrackProcessing to ${discoveryNodes}`)
+  return sampleSize(discoveryNodes, NUM_DISCOVERY_NODES_FOR_CONSENSUS)
+}
+
+async function getTrendingTracks (trendingExperiment, discoveryNodes) {
+  const results = await Promise.all(discoveryNodes.map(async node => {
+    try {
+      // The owner info is then used to target listenCount milestone notifications
+      let params = new URLSearchParams()
+      params.append('time', TRENDING_TIME.WEEK)
+      params.append('limit', MAX_TOP_TRACK_RANK)
+
+      const baseUrl = `${node}/v1/full/tracks/trending`
+      const url = trendingExperiment
+        ? `${baseUrl}/${trendingExperiment}`
+        : `${baseUrl}`
+
+      const trendingTracksResponse = await axios({
+        method: 'get',
+        url,
+        params,
+        timeout: 10000
+      })
+      const trendingTracks = trendingTracksResponse.data.data.map((track, idx) => ({
+        trackId: decodeHashId(track.id),
+        rank: idx + 1,
+        userId: decodeHashId(track.user.id)
+      }))
+      const blocknumber = trendingTracksResponse.data.latest_indexed_block
+      return { trendingTracks, blocknumber }
+    } catch (err) {
+      logger.error(`Unable to fetch trending tracks: ${err}`)
+      return null
+    }
+  }))
+
+  // Make sure we had no errors
+  if (results.some(res => res === null)) {
+    logger.error(`Unable to fetch trending tracks from all nodes`)
     return null
   }
+
+  // Make sure trending is consistent between nodes
+  const { trendingTracks, blocknumber } = results[0]
+  for (const result of results.slice(1)) {
```

Comment:
only a few fields, but a reasonable change nonetheless to only compare ids

Reply:
and does rank matter as well im guessing? so how you have is prob perf

Reply:
Ehh yeah I guess rank does matter -- though I think this should normally be in the same order since trending itself is ordered. isEqual is safe enough, no nested fields and there's only 4 of em.